### ISSUE TITLE
C14 backend support

### ISSUE THREAD

**Issue by skobkin** [duplicati/duplicati#3476]
- [ x ] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info
- **Duplicati version**: 2.0
- **Operating system**: *
- **Backend**: C14

## Description
I'm proposing to add [C14](https://www.online.net/en/c14) storage backend support.
It's cheap archive storage service from online.net that can probably can be used by duplicati.

Is has [API](https://console.online.net/en/api/) and [CLI](https://github.com/scaleway/c14-cli) interfaces.

**Comment by skobkin** [comment]
Looks like at least part of the needed functionality is probably already supported by duplicati.

C14 allows to use [FTP, SFTP, Rsync, SCP and WebDAV](https://documentation.online.net/en/c14/tutorials/start) for operations with temporary storage (which is used before archiving data to the cold storage).

But for safe (temp storage) and archive management (rotation?) some new C14-specific code probably must be needed.

**Comment by blu-e-man** [comment]
Is there anything new? 

The normal S3 storage under Scaleway works.
I tried to use the C14 storage (Glacier), but while checking the data the following error occurs:
2020-04-17 14:00:16 +02 - [Error-Duplicati.Library.Main.Operation.TestHandler-FailedToProcessFile]: Failed to process file duplicati-20200417T115913Z.dlist.zip.aes

**Comment by skobkin** [comment]
@blu-e-man 
> I tried to use the C14 storage (Glacier)

C14 is not Glacier-compliant. It's an alternative for Amazon Glacier with it's own API. That's why you can store your data on Scaleway's S3 bucket but can't on C14. You'll need a new C14 storage back-end support in Duplicati to achieve that.

**Comment by blu-e-man** [comment]
@skobkin 


Do you know if there are plans to implement c14?

**Comment by skobkin** [comment]
@blu-e-man No.

My information above was outdated. I've checked Scaleway site and found that they've released another option called "C14 Cold Storage" (I was talking about "C14 classic":

> The C14 Cold Storage is 100% S3 compliant, while the C14 Classic uses legacy FTP protocols.

So you should probably check that you're using new version instead of "Classic".

**Comment by ts678** [comment]
[Data Migration C14 Classic](https://www.scaleway.com/en/docs/data-migration-c14-classic/) says Classic is going end of life soon, so picture changes.
[C14 Cold Storage](https://www.scaleway.com/en/c14-cold-storage/) claims to be 100% S3 compatible, so no more FTP as Classic uses.
Restoration time of usually a few minutes suggests it's not just repriced hot storage.

So maybe this works now, unless the [Amazon Glacier API](https://docs.aws.amazon.com/amazonglacier/latest/dev/amazon-glacier-api.html) is required. Anyone know?
Maybe supporting C14 Cold Storage turns into [Amazon Glacier support #701](https://github.com/duplicati/duplicati/issues/701) now?

**Comment by skobkin** [comment]
Yes, now Scaleway provides C14 as S3-compatible service and you can create GLACIER objects using S3.
So it's just a matter of storing most of the files as GLACIER objects when using S3.

**Comment by arksilva** [comment]
Tested it for a bit. Backup to C14 works fine with GLACIER class selected. 
Restoring, however, does not work out of the box. Duplicati does not automatically unfreeze remote files (transfer to standard class) and since Scaleway C14 doesn't allow you to download GLACIER files directly, it throws error. So you need to manually unfreeze necessary files first.
For partial restores you might want to run restore, get an error with a list of remote file names, feed it to something like s3cmd to unfreeze them and restore again. Or maybe use --run-script-before option to automate?

**Comment by kenkendk** [comment]
Closing this as the way forward for Scaleway is S3 compatible API, which already works.
(If something does not work there, create a new issue)
